library/std/src/os/unix/fs.rs

```diff
@@ -149,7 +149,14 @@ pub trait FileExt {
     /// Note that similar to [`File::write`], it is not an error to return a
     /// short write.
     ///
+    /// # Bug
+    /// On some systems, due to a [bug] with [`pwrite64`] (the underlying
+    /// syscall), files opened with the `O_APPEND` flag fail to respect the
+    /// offset parameter, always appending to the end of the file instead.
+    ///
     /// [`File::write`]: fs::File::write
+    /// [`pwrite64`]: https://man7.org/linux/man-pages/man2/pwrite.2.html
+    /// [bug]: https://man7.org/linux/man-pages/man2/pwrite.2.html#BUGS
```

Comment:
This could use explaining in terms of Rust:
```rust
    let file = File::options().append(true).open("/tmp/foo.txt")?;
```
This problematic example is written in Rust and doesn't involve `O_APPEND` anywhere in sight. Note: since this function is platform-specific, describing the platform-specific issue and mentioning platform-specific flags is *fine*, but the programmer needs to know "yes, you can accidentally set `O_APPEND` using Rust code".